packages/@ember/-internals/runtime/lib/mixins/array.d.ts

```diff
@@ -103,5 +103,8 @@ export function A<A>(arr: A): A extends Array<infer V> ? NativeArray<V> : Native
 export function A<T>(): NativeArray<T>;
 
 export function removeAt<T>(array: T[] | EmberArray<T>, start: number, len: number): EmberArray<T>;
-export function uniqBy<T>(array: T[], keyOrFunc: string | ((item: T) => unknown)): T[];
-export function isArray(obj: unknown): boolean;
+export function uniqBy<T>(
+  array: T[] | EmberArray<T>,
+  keyOrFunc: string | ((item: T) => unknown)
+): T[] | EmberArray<T>;
+export function isArray(obj: unknown): obj is ArrayLike<unknown> | EmberArray<unknown>;
```

Comment:
Looking at the underlying code I believe this is pretty similar to what actually happens. TBH we probably want to stop using this method sooner rather than later.